src/index.ts

```diff
@@ -291,6 +291,20 @@ const main: JupyterFrontEndPlugin<IDebugger> = {
       }
     });
 
+    commands.addCommand(CommandIDs.terminate, {
+      label: 'Terminate',
+      caption: 'Terminate',
+      iconClass: 'jp-MaterialIcon jp-StopIcon',
+      isEnabled: () => {
+        return service.isThreadStopped();
+      },
+      execute: async () => {
+        await service.stop();
+        await service.start();
```

Comment:
This could actually be moved to a `service.terminate()` method, which will also contain the logic below (cleaning `clearModel`, `_stoppedThreads` and so on)

Reply:
Or better: `service.restart()` as this is what it is doing at the moment.

Reply:
Indeed that is a better name. I think we should wait for the execution to complete before restarting (or at least before sending the breakpoints back). Otherwise the code may stop again if a computation is long enough so that the sevice has the time to set the breakpoints again.

Reply:
Sending the breakpoints after restart, but without explicitly waiting for the `execute_reply`:
 
![stop-button-2](https://user-images.githubusercontent.com/591645/68309728-21871880-00af-11ea-89aa-91a4835d6cd5.gif)